### PR TITLE
WIP Bundle fonts with web assets #27

### PR DESCRIPTION
**Pull request by mazhara**

### CODE REVIEW

File: fonts.sc [hunk 1]
```diff
@@ -0,0 +1,16 @@
+import mill._
+import ammonite.ops._
+
+
+trait FontDownloader extends Module {
+
+  def getFonts = T {
```
Comment (zainab-ali) [inline fonts.sc:7]: I think we can restructure the code a bit.
Could you move this function to the `WebpackModule` trait in `webpack.sc`? That's where the web assets and stylesheets are dealt with, so it makes sense to handle fonts there too. I don't think we need a `FontDownloader` trait as it's a single function and we aren't using it in multiple places.

This should return a `PathRef` for where the fonts are downloaded. You can see examples of `PathRefs` in the `WebpackModule`, but I think the following code should do the trick:

```scala
def fonts = T {
  val dest = T.ctx().dest
  val fontsDir = dest / 'fonts
  ...
  PathRef(fontsDir)
}
```

Once you start changing the css to use `@fontface` you'll have a better idea of what this path should be.

I've renamed it from `getFonts` to `fonts` to fit with the existing code, but the name isn't too important.

You can test the download with `mill web.fonts`.

File: build.sc [hunk 2]
```diff
@@ -96,10 +97,16 @@ object server extends ScalaModule {
     millSourcePath / "src" / "main" / "resources" / "localkey.pkcs12"
   }
 
+
+  override def compile = T {
```
Comment (zainab-ali) [inline build.sc:101]: Downloading the fonts and compiling the server are two unrelated tasks (a failed download shouldn't fail compilation, for example). I think it makes sense to move this to `bundle` within the `Webpack` trait:


```scala
def bundle = T {
  vendor().map { p => os.copy.into(p.path, T.ctx.dest) }
  os.copy.into(sass().path, T.ctx.dest)
  os.copy.into(fonts().path, T.ctx.dest)
  ...
  PathRef(T.ctx.dest)
}
```

**Review by zainab-ali — CHANGES_REQUESTED** [review]
Thanks @mazhara ! This is pretty exciting to see :smiley: 

I know you're not finished with this yet, but I hope you don't mind some initial comments.

**Comment by zainab-ali** [comment]
Thanks @mazhara ! There seem to be a few other changes in the PR, possibly due to the merge. 

However, this is nothing to worry about :slightly_smiling_face:. I'll look into it and walk you through the git commands to fix it.

**Comment by zainab-ali** [comment]
I’m not quite sure what’s happened with your git history. For whatever reason, it’s diverged from the history of the upstream repository. The best way to fix it is to create a new `task27` branch and commit your changes onto it.

I’ve made a copy of your branch myself, so don’t worry if you make mistakes.

I’m assuming that your remotes are named origin and upstream:

```
git remote -v 

 > origin    git@github.com:mazhara/lsug-website.git (fetch) 
 > origin    git@github.com:mazhara/lsug-website.git (push)  
 > upstream  git@github.com:lsug/lsug-website.git (fetch)    
 > upstream  git@github.com:lsug/lsug-website.git (push)     

```

1. First, check you’re on branch `task27`.

   ```
   git branch --show-current

   > task27
   ```
2. Rename the branch to `task27-backup`.
   ```
   git branch -m task27 task27-backup
   ```

3. Switch to the master branch.
   ```
   git checkout master
   ```

4. Get it up to date with the main lsug-website repository.
   ```
   git pull upstream master
   ```

5. Checkout a new `task27` branch. This is unrelated to the previous `task27`, which has been renamed.
   ```
   git checkout -b task27
   ```

6. Checkout the `webpack.sc` file from your old branch. 
   ```
   git checkout task27-backup webpack.sc
   ```
   I think this is the only file you wish to change.

7. If you take a look at `git status`, you’ll see that the font changes from your previous branch are staged.

   ```
   git status

   > On branch task27
   > Changes to be committed:
   >   (use "git restore --staged <file>..." to unstage)
   > 	modified:   webpack.sc
   ```
8. Commit those changes.
   ```
   git commit -m "Bundle fonts with web assets"
   ```

9. Replace the `task27` branch on your origin with the cleaned up one.
   ```
   git push --force origin task27
   ```
   
   The `--force` flag is necessary as we're overwriting the contents of `origin/task27`. You shouldn't usually use it.
   
You can find out what these commands do using `man` and `--help` e.g

```
man git-checkout
git checkout --help
```

Good luck! :smiley: 
 Let me know if you have any issues.

**Comment by yilinwei** [comment]
LGTM :+1:  Thanks @mazhara. @zainab-ali do you mind double-checking that the assets get bundled in the correct locations? We can open a second issue for the `index.html` to use the correct assets.

**Comment by zainab-ali** [comment]
I'm a bit behind, but I'm really excited to take a look at this :smiley: . I'll give it a review tomorrow. Thanks very much @mazhara !

**Comment by zainab-ali** [comment]
Thanks @mazhara ! This looks perfect :smile:

**Pull request merged**